### DISCUSSION TITLE
anyone successful using mypy with pulp?

### DISCUSSION THREAD

**Discussion by epogrebnyak** [coin-or/pulp#364] [Ideas]
I'm trying to introdiuce a type alias

```
LpExpression = pulp.pulp.LpAffineExpression 
```

, but on check I get:

```
error: Variable "pkg.LpExpression" is not valid as a type
```

Is there anything I can do differently to prevent error?

Reference: https://mypy.readthedocs.io/en/latest/common_issues.html#variables-vs-type-aliases

**Comment by pchtsp** [comment]
I'm not at all famliar with mypy but: why are you doing `pulp.pulp.LpAffineExpression`? Shouldn't you be using `pulp.LpAffineExpression`?

**Reply by epogrebnyak** [reply]
same behavior for `pulp.LpAffineExpression` unfortunately

**Reply by epogrebnyak** [reply]
double pulp comes from:

```
type(pulp.lpSum(1))
Out[10]: pulp.pulp.LpAffineExpression
```
